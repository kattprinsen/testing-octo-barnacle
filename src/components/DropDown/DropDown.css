.dropdown-container {
  width: 12em;
  margin: 0 auto;
  cursor: pointer;
}

.dropdown-header {
  margin-bottom: 0.8em;
  padding: 0.4em 2em 0.4em 1em;
  box-shadow: 0 2px 3px rgba(0, 0, 0, 0.15);
  font-weight: 500;
  font-size: 1.2rem;
  color: darkgrey;
  background: #fff;
}

.dropdown-list {
  padding: 0;
  margin: 0;
  padding-left: 1em;
  background: #fff;
  border: 2px solid #e5e5e5;
  box-sizing: border-box;
  color: darkgrey;
  font-size: 1.3rem;
  font-weight: 500;
  &:first-child {
      padding-top: 0.8em;
  }
}

.dropdown-list-item {
  list-style: none;
  margin-bottom: 0.8em;
}
