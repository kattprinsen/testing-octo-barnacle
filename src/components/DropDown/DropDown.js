import React, { useState } from 'react';
import './DropDown.css';

const DropDown = ({ title, header, items }) => {

  const [isOpen, setIsOpen] = useState(false);
  const toggling = () => setIsOpen(!isOpen);
  const [selectedOption, setSelectedOption] = useState(null);

  const onOptionClick = value => () => {
    setSelectedOption(value);
    setIsOpen(false);
  }

  return (
    <div className="dropdown-container">
      <span className="dropdown-title"> {title} </span>
      <div className="dropdown-header" onClick={toggling}>
      {selectedOption || `${header}`}
      </div>
      {isOpen && (
        <div className="dropdown-list-container">
          <ul className="dropdown-list">
          {items.map(item => (
            <li className="dropdown-list-item" onClick={onOptionClick(item)} key={Math.random()}>
              {item}
            </li>
          ))}
          </ul>
        </div>
      )}
    </div>
  );
}

export default DropDown;
