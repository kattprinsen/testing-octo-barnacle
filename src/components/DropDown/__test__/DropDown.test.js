import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import DropDown from '../DropDown';

test('renders the component', () => {
  render(<DropDown />);
});

test('renders title of the dropdown', () => {
  render(<DropDown title={'Country'} />);
  expect(screen.getByText('Country')).toBeInTheDocument();
  expect(screen.getByTestId('title-id')).toHaveTextContent('Country');
});

test('renders header of the dropdown', () => {
  render(<DropDown header={'Select a country'} />);
  expect(screen.getByText('Select a country')).toBeInTheDocument();

  expect(screen.getByTestId('header-id')).toHaveTextContent('Select a country');
});

test('renders header and title', () => {
  render(<DropDown title={'Title'} header={'Header'} />);
  expect(screen.getByText('Title')).toBeInTheDocument();
  expect(screen.getByText('Header')).toBeInTheDocument();

  expect(screen.getByTestId('title-id')).toHaveTextContent('Title');
  expect(screen.getByTestId('header-id')).toHaveTextContent('Header');
});

test('click the dropdown', () => {
  render(<DropDown header={'Select a country'} />);
  userEvent.click(screen.getByTestId('header-id'));
  //Assert
});
