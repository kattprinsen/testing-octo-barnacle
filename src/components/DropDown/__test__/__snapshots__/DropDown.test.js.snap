// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dropdown-container"
>
  <span
    className="dropdown-title"
    data-testid="title-id"
  >
     
    DropDown
     
  </span>
  <div
    className="dropdown-header"
    data-testid="header-id"
    onClick={[Function]}
  >
    undefined
  </div>
</div>
`;
